pkg/mcp/sink/client_sink.go

```diff
@@ -33,62 +35,62 @@ var (
 // Client implements the client for the MCP source service. The client is the
 // sink and receives configuration from the server.
 type Client struct {
-	// Client receives configuration using the ResourceSource RPC service
-	stream mcp.ResourceSource_EstablishResourceStreamClient
-
 	client mcp.ResourceSourceClient
 	*Sink
-	reporter monitoring.Reporter
 }
 
 // NewClient returns a new instance of Client.
 func NewClient(client mcp.ResourceSourceClient, options *Options) *Client {
 	return &Client{
-		Sink:     New(options),
-		reporter: options.Reporter,
-		client:   client,
+		Sink:   New(options),
+		client: client,
 	}
 }
 
 var reconnectTestProbe = func() {}
 
 func (c *Client) Run(ctx context.Context) {
-	// The first attempt is immediate.
-	retryDelay := time.Nanosecond
+	var err error
+	var stream Stream
+
+	backoffPolicy := backoff.NewExponentialBackOff()
+	backoffPolicy.InitialInterval = time.Nanosecond
+	backoffPolicy.MaxElapsedTime = 0
+	bo := backoff.WithContext(backoffPolicy, ctx)
 
 	for {
 		// connect w/retry
-		for {
+		err = backoff.Retry(func() error {
 			select {
 			case <-ctx.Done():
-				return
-			case <-time.After(retryDelay):
+				return backoff.Permanent(fmt.Errorf("MCP sink stopped"))
+			default:
```

Comment:
You should be able to use replace line 65 in the orignial code with a read from the channel returned by `https://godoc.org/github.com/cenkalti/backoff#NewTicker`. The rest of the code would then stay the same.

Reply:
done